src/insights_client/metrics.py

```diff
@@ -51,25 +54,34 @@ def __init__(
         rhsm_cfg = configparser.ConfigParser()
         rhsm_cfg.read(rhsm_config_file)
 
-        rhsm_server_hostname = rhsm_cfg.get("server", "hostname")
-        rhsm_server_port = rhsm_cfg.get("server", "port")
-        rhsm_rhsm_repo_ca_cert = rhsm_cfg.get("rhsm", "repo_ca_cert")
-        rhsm_rhsm_consumerCertDir = rhsm_cfg.get("rhsm", "consumerCertDir")
-
-        if cert_file is None:
-            cert_file = os.path.join(rhsm_rhsm_consumerCertDir, "cert.pem")
-        if key_file is None:
-            key_file = os.path.join(rhsm_rhsm_consumerCertDir, "key.pem")
-
-        match = re.match("subscription.rhsm(.stage)?.redhat.com", rhsm_server_hostname)
-        if match is None:
-            # Assume Satellite-managed and configure for Satellite-proxied access
-            self.base_url = rhsm_server_hostname
-            self.port = rhsm_server_port
-            self.cert = (cert_file, key_file)
-            self.verify = rhsm_rhsm_repo_ca_cert
-            self.api_prefix = "/redhat_access/r/insights/platform"
-        else:
+        try:
+            rhsm_server_hostname = rhsm_cfg.get("server", "hostname")
+            rhsm_server_port = rhsm_cfg.get("server", "port")
+            rhsm_rhsm_repo_ca_cert = rhsm_cfg.get("rhsm", "repo_ca_cert")
+            rhsm_rhsm_consumerCertDir = rhsm_cfg.get("rhsm", "consumerCertDir")
+
+            if cert_file is None:
+                cert_file = os.path.join(rhsm_rhsm_consumerCertDir, "cert.pem")
+            if key_file is None:
+                key_file = os.path.join(rhsm_rhsm_consumerCertDir, "key.pem")
+
+            match = re.match("subscription.rhsm(.stage)?.redhat.com", rhsm_server_hostname)
+            if match is None:
+                # Assume Satellite-managed and configure for Satellite-proxied access
+                self.base_url = rhsm_server_hostname
+                self.port = rhsm_server_port
+                self.cert = (cert_file, key_file)
+                self.verify = rhsm_rhsm_repo_ca_cert
+                self.api_prefix = "/redhat_access/r/insights/platform"
+                is_satellite = True
+            else:
+                is_satellite = False
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            # cannot read RHSM conf, go straight to insights-client conf parsing
+            # set is_satellite=False to enter the insights-client.conf parsing block
+            is_satellite = False
+
+        if not is_satellite:
```

Comment:
The exceptions can be raised only by _rhsm_cfg.get_ on the first four lines. Let’s _try_ only those and use _else_ for the rest of the code that should not run if an exception occurs.

```suggestion
        try:
            rhsm_server_hostname = rhsm_cfg.get("server", "hostname")
            rhsm_server_port = rhsm_cfg.get("server", "port")
            rhsm_rhsm_repo_ca_cert = rhsm_cfg.get("rhsm", "repo_ca_cert")
            rhsm_rhsm_consumerCertDir = rhsm_cfg.get("rhsm", "consumerCertDir")
        except (configparser.NoSectionError, configparser.NoOptionError):
            # cannot read RHSM conf, go straight to insights-client conf parsing
            # set is_satellite=False to enter the insights-client.conf parsing block
            is_satellite = False
        else:
            if cert_file is None:
                cert_file = os.path.join(rhsm_rhsm_consumerCertDir, "cert.pem")
            if key_file is None:
                key_file = os.path.join(rhsm_rhsm_consumerCertDir, "key.pem")

            match = re.match("subscription.rhsm(.stage)?.redhat.com", rhsm_server_hostname)
            if match is None:
                # Assume Satellite-managed and configure for Satellite-proxied access
                self.base_url = rhsm_server_hostname
                self.port = rhsm_server_port
                self.cert = (cert_file, key_file)
                self.verify = rhsm_rhsm_repo_ca_cert
                self.api_prefix = "/redhat_access/r/insights/platform"
                is_satellite = True
            else:
                is_satellite = False
```

Reply:
I believe this suggestion will also fix the case where `proxy_username` is not only set to an empty string, but also when the key is absent entirely.